### PR TITLE
[SW-2408] Add H2O Isolation Forest to Algo API

### PR DESCRIPTION
**Pull request by mn-mikke**

### CODE REVIEW

File: ml/src/main/scala/ai/h2o/sparkling/ml/algos/ProblemType.scala [hunk 1]
```diff
@@ -27,7 +27,7 @@ object ProblemType extends Enumeration {
   def familyToProblemType(family: String): ProblemType = {
     val enumValue = Family.valueOf(family)
     enumValue match {
-      case Family.AUTO => Both
+//      case Family.AUTO => Both
```
Comment (jakubhava) [inline ml/src/main/scala/ai/h2o/sparkling/ml/algos/ProblemType.scala:30]: Should be uncommented

File: r/src/R/ai/h2o/sparkling/ml/models/H2OMOJOModel.R [hunk 2]
```diff
@@ -79,3 +79,10 @@ H2OTreeBasedSupervisedMOJOModel <- setRefClass("H2OTreeBasedSupervisedMOJOModel"
 #' @export H2OUnsupervisedMOJOModel
 H2OUnsupervisedMOJOModel <- setRefClass("H2OUnupervisedMOJOModel", contains = ("H2OMOJOModel"), fields = list(jmojo = "ANY"), methods = list(
 ))
+
+#' @export H2OTreeBasedUnsupervisedMOJOModel
+H2OTreeBasedUnsupervisedMOJOModel <- setRefClass("H2OTreeBasedUnsupervisedMOJOModel", contains = ("H2OUnsupervisedMOJOModel"), methods = list(
+  getNtrees = function() {
+    invoke(.self$jmojo, "getNtrees")
+  }
+))
```
Comment (jakubhava) [inline r/src/R/ai/h2o/sparkling/ml/models/H2OMOJOModel.R:88]: nit: nl

File: ml/src/main/scala/ai/h2o/sparkling/ml/algos/ProblemType.scala (hunk 1 above)
Comment (mn-mikke) [reply ml/src/main/scala/ai/h2o/sparkling/ml/algos/ProblemType.scala]: Thanks!

File: r/src/R/ai/h2o/sparkling/ml/models/H2OMOJOModel.R (hunk 2 above)
Comment (mn-mikke) [reply r/src/R/ai/h2o/sparkling/ml/models/H2OMOJOModel.R:88]: Thanks!

**Pull request merged**